### PR TITLE
[codex] Fix empty heatmap cells leaking splitArea background

### LINKED ISSUE #21290: [Bug] 热力图对于空数据的表现不一致
### Version

6.0.0

### Link to Minimal Reproduction

https://codepen.io/gicocqqo-the-lessful/pen/OPMVYzw

### Steps to Reproduce

1-9
2-9
的数据都是空，但是在图表上的表现不一样，一个有背景色，一个无背景色

### Current Behavior

在进行热力图开发时遇到的问题

### Expected Behavior

应该表现一致

### Environment

```markdown
- OS:
- Browser:
- Framework:
```

### Any additional comments?

_No response_

### PR DESCRIPTION
**Pull request by susiwen8**
## Summary
- render placeholder heatmap cells for empty values so splitArea bands do not leak through and create inconsistent empty-cell backgrounds
- add a regression test covering the empty heatmap splitArea behavior

## Root Cause
Empty heatmap cells on cartesian-style heatmaps were skipped entirely. With `splitArea` enabled, the alternating axis background showed through those skipped cells, so adjacent empty cells could appear inconsistent.

## Validation
- `./node_modules/.bin/jest --config test/ut/jest.config.cjs --runInBand test/ut/spec/series/heatmap.test.ts`
- `./node_modules/.bin/eslint src/chart/heatmap/HeatmapView.ts test/ut/spec/series/heatmap.test.ts`

Fixes #21290

### CODE REVIEW

File: src/chart/heatmap/HeatmapView.ts [hunk 1]
```diff
@@ -130,11 +131,12 @@ class HeatmapView extends ChartView {
         this.group.removeAll();
 
         const coordSys = seriesModel.coordinateSystem;
+        const emptyCellFill: ZRColor = ecModel.get('backgroundColor') || tokens.color.neutral00;
```
Comment (Ovilia) [inline src/chart/heatmap/HeatmapView.ts]: It should probably use `tokens.color.background` instead.

**Review by Ovilia — CHANGES_REQUESTED** [review]
Can you please provide a screenshot of the colors before and after the change?

**Comment by susiwen8** [comment]
Updated the fallback color in \. I also re-ran the case locally with a high-contrast \ palette and a yellow chart background to make the visual difference obvious: before the fix, the empty cells showed the split-area colors directly; after the fix, the empty cells use the chart background instead.

File: src/chart/heatmap/HeatmapView.ts (hunk 1 above)
Comment (susiwen8) [reply src/chart/heatmap/HeatmapView.ts]: Updated in `adb71c1` to use `tokens.color.background` as the fallback instead of `tokens.color.neutral00`.

**Comment by susiwen8** [comment]
Updated the fallback color in `adb71c1`.

I also re-ran the case locally with a high-contrast `splitArea.areaStyle.color` palette and a yellow chart background to make the visual difference obvious:
- before the fix, the empty cells showed the split-area colors directly
- after the fix, the empty cells use the chart background instead

**Comment by susiwen8** [comment]
image
image
